lib/yggdrasil/player.ex

```diff
@@ -12,9 +13,16 @@ defmodule Yggdrasil.Player do
     Supervisor.add_player user_id, [owner_pid, push_msg]
   end
 
-  def run_cmd(user_id, cmd = %Message{type: :command}) do
-    player_pid = Registry.get_player user_id
-    GenServer.cast(player_pid, {:run_cmd, cmd})
+  def run_cmd(user_id, cmd) do
+    # horrible terrible temporary implementation
+    # room would eventually do all of this
+    ctxt = %Yggdrasil.Room.Context{ player: user_id }
+    ctxt = Command.execute cmd, ctxt
+
+    Enum.each ctxt.actions,
+      fn {:notify, player, message} ->
+        notify(player, message)
+      end
```

Comment:
My issue is that the context will basically need to know everything about the room so that commands can do what they want. They'll need to know all the contents in the room, whose in the room, exits for the room. The player process won't have any of that info. For me it makes sense that the room receives the command from the player process. It'll know the full context since the player is in the room.

I know the code above is just an example but it does point out one of the issues I pointed out earlier. You add the move action and a notify action, but then later the move can fail. Should the code just assume you stop at the action that fails? What if you already told the player something, or notified another person in the room? Without extra code to deal with error conditions for each action the command triggered you could have partial commands or misleading messages.

If the context is the full context of the room, then the command itself can determine what it should do. As an example, the exit does not exist, the command can produce whatever message to the player is appropriate. In the case of picking up an item, the command could check that the item exits, check that the player has the strength to pick it up, check if the item has certain properties, then decide what actions should be triggered.